test/test-applications/integrations/TestApplication.ContinuousProfiler/Exporter/OtlpOverHttpExporter.cs

```diff
@@ -0,0 +1,265 @@
+            Console.WriteLine(e);
+        }
+    }
+
+    public void ExportAllocationSamples(byte[] buffer, int read)
+    {
+        var allocationSamples = SampleNativeFormatParser.ParseAllocationSamples(buffer, read);
+        if (allocationSamples.Count == 0)
+        {
+            return;
+        }
+
+        try
+        {
+            var scopeProfiles = CreateScopeProfiles();
+
+            var lastTimestamp = allocationSamples[0].ThreadSample.TimestampNanoseconds;
+            var extendedPprofBuilder = new ExtendedPprofBuilder();
+            var profileContainer = CreateProfileContainer(extendedPprofBuilder.Profile, "allocation", lastTimestamp);
+            scopeProfiles.Profiles.Add(profileContainer);
+
+            for (var i = 0; i < allocationSamples.Count; i++)
+            {
+                var allocationSample = allocationSamples[i];
+                if (allocationSample.ThreadSample.TimestampNanoseconds != lastTimestamp)
+                {
+                    // TODO consider either putting each sample in separate profile or in one profile with min and max timestamp
+                    extendedPprofBuilder = new ExtendedPprofBuilder();
+                    lastTimestamp = allocationSample.ThreadSample.TimestampNanoseconds;
+                    profileContainer = CreateProfileContainer(extendedPprofBuilder.Profile, "allocation", lastTimestamp);
+                    scopeProfiles.Profiles.Add(profileContainer);
+                }
+
+                var sampleBuilder = CreateSampleBuilder(allocationSample.ThreadSample, extendedPprofBuilder);
+
+                sampleBuilder.SetValue(allocationSample.AllocationSizeBytes);
+                extendedPprofBuilder.Profile.Sample.Add(sampleBuilder.Build());
+            }
+
+            var resourceProfiles = CreateResourceProfiles(scopeProfiles);
+
+            var profilesData = CreateProfilesData(resourceProfiles);
+
+            var request = CreateExportProfilesServiceRequest(profilesData);
+
+            using var httpRequest = CreateHttpRequest(request);
+
+            using var httpResponse = SendHttpRequest(httpRequest, CancellationToken.None);
+
+            httpResponse.EnsureSuccessStatusCode();
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(e);
+        }
+    }
+
+    private static SampleBuilder CreateSampleBuilder(ThreadSample threadSample, ExtendedPprofBuilder extendedPprofBuilder)
+    {
+        var sampleBuilder = new SampleBuilder();
+
+        if (threadSample.SpanId != 0 || threadSample.TraceIdHigh != 0 || threadSample.TraceIdLow != 0)
+        {
+            extendedPprofBuilder.AddLink(sampleBuilder, threadSample.SpanId, threadSample.TraceIdHigh, threadSample.TraceIdLow);
+        }
+
+        for (var index = 0; index < threadSample.Frames.Count; index++)
+        {
+            var methodName = threadSample.Frames[index];
+            sampleBuilder.AddLocationId(extendedPprofBuilder.GetLocationId(methodName));
```

Comment:
The way this builder pattern is used does not require a return type at all.

Reply:
You are probably right. I would keep this for now as is, I can make some additional cleanup in follow up PRs.